tests/test_project.py

```diff
@@ -417,6 +417,65 @@ def updater(remotes):
     assert ur.tr_head_0 == v1_0
     assert ur.tr_head_1 == v2_0
 
+def test_update_head_0(west_init_tmpdir):
+    # Verify that using HEAD~0 as the revision causes west to not touch the
+    # local repo. In this case, local zephyr should remain unchanged, even
+    # though it is referenced in the new top-level manifest 'my_manifest',
+    # since it's referenced with HEAD~0. Check that a local commit remains, and
+    # that local changes are kept as is. An invalid url ('na') is used to check
+    # that west doesn't attempt to access it.
+    # Expect only Kconfiglib to be updated, since it's mentioned in the new
+    # manifest's name-allowlist.
+
+    # create local repositories
+    cmd('update')
+
+    local_zephyr = west_init_tmpdir / "zephyr"
+
+    add_commit(local_zephyr, "local commit")
+    local_commit = check_output([GIT, 'rev-parse', 'HEAD'], cwd=local_zephyr)
+
+    with open(local_zephyr / "CODEOWNERS", 'a') as f:
+        f.write("\n") # Make a local change (add a newline)
+
+    my_manifest_dir = west_init_tmpdir / "my_manifest"
+    my_manifest_dir.mkdir()
+
+    my_manifest = Path(my_manifest_dir / "west.yml")
+    my_manifest.write_text(
+        '''
+          manifest:
+            projects:
+              - name: zephyr
+                revision: HEAD~0
+                url: na
+                import:
+                  name-allowlist:
+                    - Kconfiglib
+        ''')
+
+    cmd(f"config manifest.path {my_manifest_dir}")
+
+    # Add commits to the local repos.
+    ur = update_helper(west_init_tmpdir)
```

Comment:
Before looking it up I had no idea what `update_helper()` returns. I see why you want a short name but maybe add a comment? Or maybe `up_res`

Reply:
> I see why you want a short name but maybe add a comment? Or maybe up_res

I agree that I could have picked a better name for this helper function. A comment that mentions the UpdateResults type will be sufficient for me in this case. (The other calls in this file save the return value in a `ur` variable, so I think consistency is good, and renaming the other variables is out of scope for this PR.)